src/main/java/com/cloudbees/jenkins/plugins/bitbucket/endpoints/AbstractBitbucketEndpoint.java

```diff
@@ -55,16 +66,42 @@
     @CheckForNull
     private final String credentialsId;
 
+    /**
+     * Jenkins Server Root URL to be used by that Bitbucket endpoint.
+     * The global setting from Jenkins.getActiveInstance().getRootUrl()
+     * will be used if this field is null or equals an empty string.
+     * This variable is bound to the UI, so an empty value is saved
+     * and returned by getter as such.
+     */
+    private final String bitbucketJenkinsRootUrl;
+
+    /**
+     * Value of not-empty bitbucketJenkinsRootUrl normalized for end-users
+     * and saved, to avoid recalculating it over and over; an empty or null
+     * value still causes evaluation of current global setting every time.
+     */
+    private transient String endpointJenkinsRootUrl;
+
+
     /**
      * Constructor.
      *
      * @param manageHooks   {@code true} if and only if Jenkins is supposed to auto-manage hooks for this end-point.
      * @param credentialsId The {@link StandardCredentials#getId()} of the credentials to use for
      *                      auto-management of hooks.
+     * @param bitbucketJenkinsRootUrl The custom (or empty for global setting) Jenkins Root URL
+     *                      auto-management of hooks.
      */
-    AbstractBitbucketEndpoint(boolean manageHooks, @CheckForNull String credentialsId) {
+    AbstractBitbucketEndpoint(boolean manageHooks, @CheckForNull String credentialsId, @CheckForNull String bitbucketJenkinsRootUrl) {
         this.manageHooks = manageHooks && StringUtils.isNotBlank(credentialsId);
-        this.credentialsId = manageHooks ? credentialsId : null;
+        if (this.manageHooks) {
+            this.credentialsId = credentialsId;
+            this.bitbucketJenkinsRootUrl = bitbucketJenkinsRootUrl ;
+        } else {
+            this.credentialsId = null;
+            this.bitbucketJenkinsRootUrl = "";
+        }
+        this.setEndpointJenkinsRootUrl(this.bitbucketJenkinsRootUrl);
```

Comment:
This line could be moved to a readResolve, so when Jenkins reads the XML config it will normalize and set the transient property on restart. And on set of `bitbucketJenkinsRootUrl` it should call `setEndpointJenkinsRootUrl`

Reply:
Don't know if the readResolve can be on the abstract class or it has to be in both endpoints.